### PR TITLE
Add missing notifications tests.

### PR DESCRIPTION
**Pull request by hober**
Added tests for the various events that didn't already have tests.
Added tests for the lang property (which is required to only roundtrip valid BCP47 language tags).

### CODE REVIEW

**Comment by zqzhang** [comment]
w3c-test:mirror

File: notifications/event-onclick.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+<body>
+  <div id="log"></div>
+  <script>
+var t = async_test("the notification.onclick event can be invoked", {timeout: 50000});
+
+Notification.requestPermission(t.step_func(function(permission) {
+    assert_equals(permission, "granted", "notification permission");
+
+    var notification = new Notification("Please click on the notification.", {
+        body: "Click me to test clicking on notifications."
```
Comment (zqzhang) [inline notifications/event-onclick.html]: Test needs user interaction shall be treated as a manual test. Please add `-manual` to the file name.

File: notifications/event-onerror-default.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+<link rel="author" title="Apple Inc." href="http://www.apple.com/">
+<link rel="help" title="5 API" href="http://www.w3.org/TR/notifications/">
+<meta name="flags" content="">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+  <div id="log"></div>
+  <script>
+    if (Notification.permission != "default") {
+      document.getElementById("log").innerHTML = "Please clear your notification settings for web-platform.test and retry.";
```
Comment (zqzhang) [inline notifications/event-onerror-default.html]: ditto

File: notifications/event-onerror-default.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    } else {
+    t = async_test("the notification.onerror event can be invoked", {timeout: 50000});
```
Comment (zqzhang) [inline notifications/event-onerror-default.html]: indent

File: notifications/notification-lang.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+    for (var i=0; i<well_formed_langs.length; i++) {
+        test_lang(well_formed_langs[i], false);
+    }
+
+    for (var i=0; i<invalid_langs.length; i++) {
+        test_lang(invalid_langs[i], false);
+    }
+
+    done();
```
Comment (zqzhang) [inline notifications/notification-lang.html]: This needs to setup the explicit_done.

File: notifications/event-onclick.html (hunk 1 above)
Comment (sideshowbarker) [inline notifications/event-onclick.html]: Fixed in https://github.com/w3c/web-platform-tests/commit/a8d5f0411b8816c6c4aadf9896a47f6fbcf3431b

File: notifications/event-onerror-default.html (hunk 2 above)
Comment (sideshowbarker) [inline notifications/event-onerror-default.html]: Fixed in https://github.com/w3c/web-platform-tests/commit/a8d5f0411b8816c6c4aadf9896a47f6fbcf3431b

File: notifications/event-onerror-default.html (hunk 3 above)
Comment (sideshowbarker) [inline notifications/event-onerror-default.html]: Fixed in https://github.com/w3c/web-platform-tests/commit/5fe0446befcadbd01071cbf5e6df412a587581bc

File: notifications/notification-lang.html (hunk 4 above)
Comment (sideshowbarker) [inline notifications/notification-lang.html]: Will address this over in PR #1864

**Comment by sideshowbarker** [comment]
Closing this in favor of PR #1864

**Pull request closed without merging**